plugin/session_buffer.py

```diff
@@ -265,6 +273,32 @@ def some_view(self) -> Optional[sublime.View]:
             return sv.view
         return None
 
+    def _if_view_unchanged(self, f: Callable[[sublime.View, Any], None], version: int) -> Callable[[Any], None]:
+        """
+        Ensures that the view is at the same version when we were called, before calling the `f` function.
+        """
+        def handler(*args: Any) -> None:
+            view = self.some_view()
+            if view and view.change_count() == version:
+                f(view, *args)
+
+        return handler
+
+    # --- textDocument/documentColor -----------------------------------------------------------------------------------
+
+    def _do_color_boxes_async(self, view: sublime.View, version: int) -> None:
+        if self.session.has_capability("colorProvider"):
+            self.session.send_request_async(
+                Request.documentColor(document_color_params(view), view),
+                self._if_view_unchanged(self._on_color_boxes_async, version)
```

Comment:
Currently the request will be send regardless if the view is unchanged. 
I think it would be a better to send the `documentColor` request only when the view is unchanged. (to avoid causing unnecessary tragic to the server)

----
Currently `_do_color_boxes_async` is called from `purge_changes_async`,
`purge_changes_async` is called in `on_text_changed_async`, 
and `purge_changes_async` is debounced there, 
so we won't hit the problem I described above.

Reply:
`_do_color_boxes_async` is only triggered after didOpen and didChange notifications so that implies that the view just changed and is stable. Isn't it?

Reply:
_do_color_boxes_async is also called from purge_changes_async.

purge_changes_async is called in on_text_changed_async,
But purge_changes_async is debounced there, so we won't hit the problem I described above...
Still I think it would be preferred to not send the request at all until the vue remains unchanged.

What do you think?

Reply:
> _do_color_boxes_async is also called from purge_changes_async.

That's where the `didChange` notification is sent and it's only sent when there were pending changes that we just purged and at this point the change_count() is at latest state.

So again, I'm not getting what you mean really...

Reply:
Hahaha, I wrote vue instead of view... what a brain bug 😀

Reply:
> So again, I'm not getting what you mean really...

Is there a need to handle the response only when the view remained unchanged? 
Can this line could be changed  to:
```diff
            self.session.send_request_async(
                Request.documentColor(document_color_params(view), view),
-                self._if_view_unchanged(self._on_color_boxes_async, version)
+                self._on_color_boxes_async
```

It seems unnecessary to fire the request and to handle response **only** when the view is unchanged:

```py
            self.session.send_request_async(
                Request.documentColor(document_color_params(view), view),
                self._if_view_unchanged(self._on_color_boxes_async, version) #  handle the response only if the view remained unchanged
```

Reply:
The view is unchanged at the point of making the request. That can change by the time the response arrives so we have to check.